pkg/lease/interface.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lease
+
+import "time"
+
+// Lease proxies to an auto-expiring lease stored in a LeaseRegistry.
+// The creator of a Lease must repeatedly call Renew to keep their lease
+// from expiring.
+type Lease interface {
+	// Renew attempts to extend the Lease TTL to the provided duration.
+	// The operation will succeed only if the Lease has not changed in
+	// the LeaseRegistry since it was last renewed or first acquired.
+	// An error is returned if the Lease has already expired, or if the
+	// operation fails for any other reason.
+	Renew(time.Duration) error
+
+	// Release relinquishes the ownership of a Lease back to the Registry.
+	// After calling Release, the Lease object should be discarded. An
+	// error is returned if the Lease has already expired, or if the
+	// operation fails for any other reason.
+	Release() error
+
+	// MachineID returns the ID of the Machine that holds this Lease. This
+	// value must be considered a cached value as it is not guaranteed to
+	// be correct.
+	MachineID() string
+
+	// Version returns the current version at which the lesse is operating.
```

Comment:
@jonboulle good catch - @ecnahc515 if you can fix this one last typo and I'll hit the merge button.